agent/src/main.rs

```diff
@@ -13,6 +13,23 @@ use std::sync::Arc;
 
 const PROG: &str = "crucible-agent";
 const SERVICE: &str = "oxide/crucible/downstairs";
+/*
+ * As a safety mechanism to prevent the agent from creating more regions
+ * than we have physical space, we using a zfs reservation on each dataset
+ * we create.  While not a complete solution, it does help in some
+ * situations to avoid allocation of a region that we don't have the
+ * space for if that region was to fill up.
+ * As for how much we reserve, it's the region size itself, plus the
+ * amount we need for metadata (currently around 17%) then an additional
+ * 8% that, if things are behaving, we will have for snapshots and
+ * to prevent us from using all the data in the pool.
+ *
+ * In addition, we throw a quota of 3x the region size.  If the region
+ * has grown that big, then something is wrong and we should prevent it
+ * from growing any larger and impacting other regions in the dataset.
+ */
+const METADATA_BUFFER: f64 = 1.25;
+const REGION_QUOTA: u64 = 3;
```

Comment:
maybe: `RESERVATION_FACTOR` and `QUOTA_FACTOR`? I leave it to you... just a thought

Reply:
Done